### PR TITLE
Tests for GithubCommands module

### PR DESCRIPTION
**Pull request by HoffsMH**
Connected to  #48, according to my coveralls this puts coverage at 57%

Also fixes a couple bugs in the module which I will leave comments next to
I took the liberty of trying to converge on a common spelling of `reusable` if there is a preferred alternate spelling let me know and I can rename.

If this is too hefty I can split out into separate PRs as well!

### CODE REVIEW

File: lib/slax/commands/github_commands.ex [hunk 1]
```diff
@@ -27,8 +33,13 @@ defmodule Slax.Commands.GithubCommands do
...
   end
 
+  @doc """
+  Pulls all issue templates from a story repo that are included in `story_paths`.
+  It  then uses these templates to create issues in a newly created github repository,
+  If there is no github repository the function will exit early
+  """
   def create_reusable_stories(
-        %{project_name: project_name} = results,
+        %{project_name: project_name, github_repo: _} = results,
```
Comment (HoffsMH) [inline lib/slax/commands/github_commands.ex]: requiring this field in the results map allows us to skip the step if there is no `github_repo` to post reusable stories to

File: lib/slax/commands/github_commands.ex [hunk 2]
```diff
@@ -70,10 +81,12 @@ defmodule Slax.Commands.GithubCommands do
         results
 
       {:error, message} ->
-        Map.update(results, :errors, %{}, fn x -> Map.put(x, :resuseable_stories, message) end)
+        Map.update(results, :errors, %{}, fn x -> Map.put(x, :reusable_stories, message) end)
     end
   end
 
+  def create_reusable_stories(results, _, _, _, _), do: results
```
Comment (HoffsMH) [inline lib/slax/commands/github_commands.ex]: The default case when no `github_repo` from a previous step exist is now to just return the results map

File: lib/slax/commands/github_commands.ex [hunk 3]
```diff
@@ -138,7 +151,7 @@ defmodule Slax.Commands.GithubCommands do
     end)
     |> Enum.split_with(fn
       {:ok, _, _} -> true
-      {:error, _} -> false
+      {:error, _, _} -> false
```
Comment (HoffsMH) [inline lib/slax/commands/github_commands.ex]: the previous `Enum.map` was generating either 2-element `:ok` tuples or 3-element `:error` tuples but `split_with` was matching on a 2-element `:error` tuple here. This resulted in `split_with` raising an error any time there was an error creating an issue.

**Comment by bryanjos** [comment]
@HoffsMH looks like there is a conflict

**Comment by HoffsMH** [comment]
@bryanjos Rebased!

**Comment by bryanjos** [comment]
Thanks!

**Pull request merged**